packages/compartment-mapper/src/node-modules.js

```diff
@@ -761,19 +761,36 @@ const graphPackages = async (
  * @param {Graph} graph
  * @param {Set<string>} conditions - build conditions about the target environment
  * for selecting relevant exports, e.g., "browser" or "node".
- * @param {SomePolicy} [policy]
+ * @param {TranslateGraphOptions} [options]
  * @returns {CompartmentMapDescriptor}
  */
 const translateGraph = (
   entryPackageLocation,
   entryModuleSpecifier,
   graph,
   conditions,
-  policy,
+  { policy, additionalPackageDetails = {} } = {},
 ) => {
   /** @type {CompartmentMapDescriptor['compartments']} */
   const compartments = create(null);
 
+  for (const [additionalPackageLocation, moduleDetails] of entries(
+    additionalPackageDetails,
+  )) {
+    const node = graph[additionalPackageLocation];
+    if (node === undefined) {
+      throw new Error(
+        `Cannot find package at ${additionalPackageLocation} in graph`,
+      );
+    }
+    for (const {
+      packageLocation,
+      packageDescriptor: { name },
+    } of moduleDetails) {
+      node.dependencyLocations[name] ??= packageLocation;
```

Comment:
this seems to be sufficient for associating a `Node` with another `Node`, but maybe there's a safer way to do so?